syntax = "proto3";

package lvi;

import "protos/bigquery_options.proto";
import "protos/client.proto";
import "protos/actor.proto";

message Tag {
    string tag_name = 1;
    string tag_code = 2;
    string tag_namespace = 3;
}

message BatchEvent {
    Actor actor = 1  [(required) = false, (description) = "Actor concerned with the event"];
    repeated Tag tags = 2 [(description) = "Tags associated with the event"];
}

message Event {
    option (table_root) = true;
    Client client = 1  [(required) = false, (description) = "Owner of the event"];
    repeated BatchEvent events = 2 [(required) = true, (batch_attribute) = true];
}